### PR TITLE
Upgrade k8s dependencies to v1.17.6, controller-runtime to v0.5.3

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/kind enhancement
/priority normal

**What this PR does / why we need it**:
This PR revendors k8s dependencies with version v0.17.6 (k8s version v1.17.6) and controller-runtime with version v0.5.3.
This contains bug fixes for the controller-runtime cache, which will be required for what I am currently working on.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement developer
The golang Kubernetes dependencies have been upgraded to `v1.17.6`.
```
```action operator
The minimum required version for the Garden Cluster was increased to `1.16`.
```

### CODE REVIEW

**Comment by rfranzke** [comment]
What about

>**Special notes for your reviewer**:
I didn't vendor the `kubernetes-1.17` sources yet because the controller-runtime is still using `kubernetes-1.16`. Also, there are two problems that I encountered when trying this:

> ```
> E1222 18:37:10.871848 66663 dynamic_serving_content.go:164] key failed with : missing content for serving cert "serving-cert::/dev/fd/63::/dev/fd/62"
> ```

> --> It seems not possible to disable the dynamic serving cert refresh controller for our local developments, so we maybe would have to adopt our `hack/start-*` scripts.

> ```
> E1222 18:37:11.229574 66663 reflector.go:156] k8s.io/client-go/informers/factory.go:135: Failed to list *v1.ValidatingWebhookConfiguration: the server could not find the requested resource
> E1222 18:37:11.229574 66663 reflector.go:156] k8s.io/client-go/informers/factory.go:135: Failed to list *v1.MutatingWebhookConfiguration: the server could not find the requested resource
> ```

> --> It seems that `k/apiserver` now tries to list the webhook configurations in `admissionregistration.k8s.io/v1` version which is only served in 1.16+ clusters.
> --> We would have to bump the minimum required Garden cluster version from 1.10 to 1.16, pretty steep at this point in time. :(
> --> Let's not try to revendor `kubernetes-1.17` at this point in time but separately at a later point in time.

?

(ref https://github.com/gardener/gardener/pull/1771)

**Comment by rfranzke** [comment]
/hold
until v1.5 is released

**Comment by rfranzke** [comment]
As discussed, it's now time to bump the minimum garden cluster version to 1.16. Hence, this PR just needs adaptations in the documentation, a proper release note, plus changes in the `garden` package that checks the minimum version.

**Comment by timebertt** [comment]
I've updated the PR:
- bumped the minimum required garden cluster version (also added release note)
- adapted our `hack/start-*` scripts to copy tls cert and key from the current kubeconfig to a tmp dir and use those files explicitly for the API server's flags
- adapted usages of k8s dependencies to signature changes
- reworked namespace defaulting in `kubernetes.Applier` (based on rest mapping, similar to how I did it in https://github.com/gardener/gardener-resource-manager/pull/47)

I hope I've covered all blockers now, PTAL :)

/needs/review
/remove needs changes
/remove status author-action
/topology garden

**Comment by gardener-botmanager** [comment]
@tim-ebert Label "status" does not exist for this repository.

**Comment by gardener-botmanager** [comment]
@tim-ebert Label "topogoly/garden" does not exist for this repository.

File: pkg/operation/botanist/constraints_check.go [hunk 1]
```diff
@@ -87,10 +85,8 @@ func (b *Botanist) CheckHibernationPossible(ctx context.Context, constraint gard
 	}
 
 	for _, webhookConfig := range mutatingWebhookConfigs.Items {
-		for i, w := range webhookConfig.Webhooks {
-			uid := fmt.Sprintf("%s/%s/%d", webhookConfig.Name, w.Name, i)
-			accessor := webhook.NewMutatingWebhookAccessor(uid, webhookConfig.Name, &w)
```
Comment (rfranzke) [inline pkg/operation/botanist/constraints_check.go:92]: Why did we need this earlier?

Comment (timebertt) [reply pkg/operation/botanist/constraints_check.go:92]: Before, we used the webhook accessor to uniformly handle mutating and validating webhooks.
Now that k8s deps are bumped, `New{Mutating,Validating}WebhookAccessor` can only take webhook configs in version `admissionregistration/v1` which we cannot assume to be present in all Shoot clusters. So I just get the config in version `v1beta1` and pass the relevant fields/slices down to the helper func.

Comment (rfranzke) [reply pkg/operation/botanist/constraints_check.go:92]: Yes, understood. I'm just wondering why we didn't do it like this earlier already, and instead created the webhook accessor. Do you know?

Comment (timebertt) [reply pkg/operation/botanist/constraints_check.go:92]: I think, I introduced it back then, but with no special reason. :)

Comment (rfranzke) [reply pkg/operation/botanist/constraints_check.go:92]: OK, I see, thank you!

File: go.mod [hunk 2]
```diff
@@ -40,37 +40,37 @@ require (
...
-	k8s.io/kube-openapi v0.0.0-20191107075043-30be4d16710a
-	k8s.io/kubelet v0.16.8
-	k8s.io/metrics v0.16.8
+	k8s.io/kube-aggregator v0.17.6
+	k8s.io/kube-openapi v0.0.0-20200410145947-bcb3869e6f29
+	k8s.io/kubelet v0.17.6
+	k8s.io/metrics v0.17.6
 	k8s.io/utils v0.0.0-20200327001022-6496210b90e8
-	sigs.k8s.io/controller-runtime v0.4.0
+	sigs.k8s.io/controller-runtime v0.5.3
```
Comment (ialidzhikov) [inline go.mod]:
```suggestion
	sigs.k8s.io/controller-runtime v0.5.4
```

Comment (timebertt) [reply go.mod]: Done

**Comment by rfranzke** [comment]
Sorry @tim-ebert, you need to rebase again :(
/needs rebase

**Comment by timebertt** [comment]
Done.
/remove needs/rebase

**Review by rfranzke — APPROVED** [review]
/lgtm

**Review by timuthy — APPROVED** [review]

**Pull request merged**